DREAM/Optimization/tsgParticleSwarm.hpp

```diff
@@ -86,53 +86,63 @@ class ParticleSwarmState {
     //! \brief Return the number of particles.
     inline int getNumParticles() const {return num_particles;}
     //! \brief Return the particle positions.
+    inline void getParticlePositions(double pp[]) const {std::copy_n(particle_positions.begin(), num_particles * num_dimensions, pp);}
     inline std::vector<double> getParticlePositions() const {return particle_positions;}
     //! \brief Return the particle velocities.
+    inline void getParticleVelocities(double pv[]) const {std::copy_n(particle_velocities.begin(), num_particles * num_dimensions, pv);}
     inline std::vector<double> getParticleVelocities() const {return particle_velocities;}
     //! \brief Return the previously best known particle positions.
+    inline void getBestParticlePositions(double bpp[]) const {std::copy_n(best_particle_positions.begin(), (num_particles + 1) * num_dimensions, bpp);}
     inline std::vector<double> getBestParticlePositions() const {return best_particle_positions;}
     //! \brief Return the previously best known position in the swarm.
+    inline void getBestPosition(double bp[]) const {std::copy_n(best_particle_positions.begin() + num_particles * num_dimensions, num_dimensions, bp);}
     inline std::vector<double> getBestPosition() const {
         std::vector<double> best_position(num_dimensions);
-        std::copy_n(best_particle_positions.begin() + num_particles * num_dimensions, num_dimensions, best_position.begin());
+        getBestPosition(best_position.data());
         return best_position;
     }
     //! \brief Return the state vector.
+    inline void getStateVector(bool sv[]) const {
```

Comment:
Is this ever used? The returned `std::vector<bool>` is used only for testing, testing is fine, but why is this needed?

If you want to expose the state to the user, you should use expressive call, e.g., `isPositionInitialized()`.

Reply:
I decided to make additional functions (`isPositionInitialized()`, `isVelocityInitialized()`, `isBestPositionInitialized()`, `isCacheInitialized()`)  to expose these states to the user.